chia/util/generator_tools.py

```diff
@@ -16,7 +16,7 @@
 def get_block_header(block: FullBlock, tx_addition_coins: List[Coin], removals_names: List[bytes32]) -> HeaderBlock:
     # Create filter
     byte_array_tx: List[bytearray] = []
-    addition_coins = tx_addition_coins + list(block.get_included_reward_coins())
+    addition_coins = tx_addition_coins + block.get_included_reward_coins()
```

Comment:
This can be totally agnostic to the types it is combining.

if you want a list:
```suggestion
    addition_coins = [*tx_addition_coins, *block.get_included_reward_coins()]
```
or a set:
```suggestion
    addition_coins = {*tx_addition_coins, *block.get_included_reward_coins()}
```

Reply:
it turns out it's not necessary to combine these collections. and in fact, most of the time it's wasted work